fdk.go

```diff
@@ -107,6 +111,84 @@ func doHTTP(handler Handler, ctx context.Context, in io.Reader, out io.Writer) {
 	}
 }
 
+func doJSON(handler Handler, ctx context.Context, in io.Reader, out io.Writer) {
+	var buf bytes.Buffer
+	hdr := make(http.Header)
+
+	for {
+		doJSONOnce(handler, ctx, in, out, &buf, hdr)
+	}
+}
+
+type jsonio struct {
+	Body        string `json:"body"`
+	ContentType string `json:"content_type"`
+}
+
+type callRequestHTTP struct {
+	Type       string      `json:"type"`
+	RequestURL string      `json:"request_url"`
+	Headers    http.Header `json:"headers"`
+}
+
+type jsonIn struct {
+	jsonio
+	CallID   string           `json:"call_id"`
+	Protocol *callRequestHTTP `json:"protocol"`
```

Comment:
ditto about pointers, this can easily be checked for `.StatusCode != 0` here for nil-ity